folding-schemes/src/folding/protogalaxy/folding.rs

```diff
@@ -306,26 +318,23 @@ where
             betas: betas_star,
             phi: phi_star,
             e: e_star,
-            u: u_star,
             x: x_star,
         })
     }
 }
 
 // naive impl of pow_i for betas, assuming that betas=(b, b^2, b^4, ..., b^{2^{t-1}})
-fn pow_i<F: PrimeField>(i: usize, betas: &[F]) -> F {
+pub fn pow_i<F: PrimeField>(i: usize, betas: &[F]) -> F {
     // WIP check if makes more sense to do it with ifs instead of arithmetic
```

Comment:
I think that now we can get rid of this old comment :)

Reply:
That makes sense! I have in fact removed the WIP comment in #145 and added docs to the function. But let me backport it to this PR, because #145 is quite huge and I don't want to take up too much of your time for the code review 😄.
https://github.com/privacy-scaling-explorations/sonobe/blob/ae71901f945bc83cf2fa93fedade5c6c49eb2979/folding-schemes/src/folding/protogalaxy/utils.rs#L74-L91

Reply:
Addressed in https://github.com/privacy-scaling-explorations/sonobe/pull/123/commits/4ab5ac0156a4b23167f97b59aa086073937baa44 :)

I also use a while loop to replace the previous `bit_decompose` call, because we can return early when `i` becomes zero, without knowing all bits in `i`.

Reply:
Nice!